### DISCUSSION TITLE
Welcome to Factor Discussions!

### DISCUSSION THREAD

**Discussion by erg** [factor/factor#2387] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Enabling the discussions with mostly the default text.

**Comment by Bubbler-4** [comment]
Hi, I'm Bubbler. I ran into Factor last month and was immediately hooked, so I started solving some challenges with it. I'm currently participating in [Advent of Code 2020](https://adventofcode.com/2020), and also going through "Bioinformatics Stronghold" track on [Rosalind](http://rosalind.info/problems/list-view/). I'm uploading the solutions [here](https://github.com/Bubbler-4/factor-problem-solving). I'm also using Factor on [Code Golf SE](https://codegolf.stackexchange.com/).

**Reply by alex-ilin** [reply]
Welcome, Bubbler!

Code Golf is kind of pointless since languages like GolfScript and Jelly were created. Rosalind is interesting.

I myself want to create log analysis tool in Factor based on the idea of scripts (as in plays or movies, not software scripts), so that I can quickly visualize what the hardware was doing and which scenarios were interrupted by errors or just simply don't look right.

**Comment by mrjbq7** [comment]
I kind of like the idea of code golf, not to compete with GolfScript, but to see where we can improve our standard library with useful higher level concepts.

Contributions welcome!

We have some rosetta-code solutions contributed in extra/rosetta-code.  If you’d like to contribute yours, that would be awesome.  Everything committed to Factor has the BSD license.

Best,
John.

**Comment by kusumotonorio** [comment]
Hi there. A few years ago, I came across Factor while looking for a suitable programming and execution environment for industrial devices with a GUI.
Factor has a unique appeal, doesn't it? It's a lot of fun. I haven't made any real gains from it so far, but it's hard to leave.😀

**Reply by alex-ilin** [reply]
It's quite cool for scripting where one needs to hack together a moderately complex tool. In Factor it'll be easy to make and will be quite concise. For example, I wrote a testing utility that reads a CSV file with some order information, then reads console input (provided by a barcode scanner on a conveyor) and generates XML order files by combining that input with the CSV data. All in a single vocab under 4Kb in size.

There is another vocab to generate the CSVs for various testing scenarios, and that one is almost 11Kb. I really like how Factor fits various data processing tasks. Kudos to whoever invented the `>>prop` and `prop>>` accessor syntax, it's very cool.

**Comment by Jimmysnielsen** [comment]
Hi !
I started grokking factor about a month ago and I find that it fits my way of coding very well. 
I recently came from Racket and also like the Lispy style of coding. 
I am working on an implementation of IAPWS-97 in factor now( a model for calculating properties of steam , pdf found here:  http://www.iapws.org/relguide/IF97-Rev.html ) 
and when I have some more code under my belt I would like to contribute to the factor base. I have already stumbled upon some bugs, but I haven't reported them as they may already have been recognised and I dont have an overview of the current open issues.
Any way, hope to be able to contribute more in the future.

Fun fact. the first time I stumbled upon factor was in this meme
https://www.reddit.com/r/lisp/comments/aqcqx/how_a_common_lisp_programmer_views_users_of_other/
Being a Star Wars fan I had it on my work pc for many years.

Fun fact2. same picture equating Forth to Ted Kaczynski made me laugh my but off! That was exactly my impression of old Forth  back in the days.

**Reply by kusumotonorio** [reply]
Hi !

Rather, Obi Wan calling into his head, "Use the FORTH, Luke." is more appropriate for Forth than Factor.
From a pure Jedi perspective, we are probably closer to Lord Vader. :-)

**Comment by mrjbq7** [comment]
Welcome!

Please forward to me any bugs that you see, or post them on GitHub.

Our intent is to improve as much as we are able to.

Best,
John.
